v2/sbomgcr/sbom_payload.go

```
@@ -0,0 +1,152 @@
+}
+
+// SBOMPredicate is an interface for the SBOM Predicates voucher supports
+type SBOMPredicate interface {
+	GetPredicateType() string
+	ToBOM(sbom *cyclonedx.BOM) error
+}
+
+// CustomPredicate represents a custom sbom predicate
+type CustomPredicate struct {
+	Type          string `json:"_type"`
+	PredicateType string `json:"predicateType"`
+	Subject       []struct {
+		Name   string `json:"name"`
+		Digest struct {
+			Sha256 string `json:"sha256"`
+		} `json:"digest"`
+	} `json:"subject"`
+	Predicate struct {
+		Data      string    `json:"Data"`
+		Timestamp time.Time `json:"Timestamp"`
+	} `json:"predicate"`
+}
+
+func (cp CustomPredicate) GetPredicateType() string {
+	return cp.PredicateType
+}
+
+func (cp CustomPredicate) ToBOM(sbom *cyclonedx.BOM) error {
+	return json.Unmarshal([]byte(cp.Predicate.Data), &sbom)
+}
+
+// CycloneDXPredicate represents a cyclonedx sbom predicate
+type CycloneDXPredicate struct {
+	Type          string `json:"_type"`
+	PredicateType string `json:"predicateType"`
+	Subject       []struct {
+		Name   string `json:"name"`
+		Digest struct {
+			Sha256 string `json:"sha256"`
+		} `json:"digest"`
+	} `json:"subject"`
+	Predicate struct {
+		Data map[string]interface{} `json:"Data"`
+	} `json:"predicate"`
+}
+
+func (cyclonedxp *CycloneDXPredicate) GetPredicateType() string {
+	return cyclonedxp.PredicateType
+}
+
+func (cyclonedxp *CycloneDXPredicate) ToBOM(sbom *cyclonedx.BOM) error {
+	return mapstructure.Decode(cyclonedxp.Predicate.Data, &sbom)
+}
+
+func getEnvelopeFromReader(reader io.ReadCloser) (Envelope, error) {
+	bt, _ := io.ReadAll(reader)
+	var envelope Envelope
+
+	err := json.Unmarshal(bt, &envelope)
+
+	if err != nil {
+		return envelope, fmt.Errorf("error unmarshalling into envelope %w", err)
+	}
+
+	return envelope, nil
+}
+
+func getSBOMFromEnvelope(envelope Envelope, cyclonedxBOM *cyclonedx.BOM) error {
+	decoded, _ := base64.StdEncoding.DecodeString(string(envelope.Payload))
```

Question: why is it safe to ignore this error? It seems spooky to me.